src/plugins/data_source_management/public/components/edit_data_source/components/edit_form/edit_data_source_form.tsx

```diff
@@ -60,8 +60,12 @@ export interface EditDataSourceState {
   description: string;
   endpoint: string;
   auth: {
-    type: AuthType;
-    credentials: UsernamePasswordTypedContent | SigV4Content;
+    type: AuthType | string;
+    credentials:
+      | UsernamePasswordTypedContent
+      | SigV4Content
+      | { [key: string]: string }
+      | undefined;
```

Comment:
is it possible for credentials to be undefined in edit page?

Reply:
Thanks for the comment, it is possible:

https://github.com/opensearch-project/OpenSearch-Dashboards/blob/99ae1a5db2b90f377ea8488bf1279cf932ac04c2/src/plugins/data_source_management/public/components/edit_data_source/components/edit_form/edit_data_source_form.tsx#L345